storage/src/main/java/org/apache/kafka/storage/internals/log/TimeIndex.java

```diff
@@ -60,6 +60,7 @@ public TimeIndex(File file, long baseOffset, int maxIndexSize) throws IOExceptio
         this(file, baseOffset, maxIndexSize, true);
     }
 
+    @SuppressWarnings("this-escape")
```

Comment:
I think this might be due to super class methods such as maxEntries(). We can make them final. AbstractIndex is not a public class [1]

(same for other index files)

[1] https://kafka.apache.org/36/javadoc/allclasses-index.html

Reply:
No, that doesn't work. It only works if the method is in the same class. It's already escaped if it's a superclass.

Reply:
Hmmm, just for my curiosity, the super class object is guaranteed to be created in the sub-class ctor because `super()` is the first line in sub-class ctor. Hence, there is no risk (of a partial object) if we call it's non-overloadable methods because the super class instance whose methods we are referring to will be completely constructed. Then why is this a warning?

(nevermind if you don't have the answer on top of your head. Java language semantics is not my strong suite so I might be missing something here)

Reply:
I think the current implementation is conservative when it comes to any assumptions for separate compilation units. After all, one can change the implementation of the superclass method to invoke a method that is overridden/implemented by the subclass without recompiling the subclass. I did test various variations and it doesn't seem possible to fix the warning while invoking any superclass methods.